spyder/api/panel.py

```diff
@@ -125,6 +131,27 @@ def paintEvent(self, event):
                 self.palette().windowText().color()))
             painter = QPainter(self)
             painter.fillRect(event.rect(), self._background_brush)
+        else:
+            raise NotImplementedError(
+                f'paintEvent method must be defined in {self}')
+
+    def sizeHint(self):
+        """
+        Return the widget size hint, override the Qt method.
+
+        Notes:
+            - This size hint will define the QSize, where the width and height
+            is defined.
+            - If the size is going to be dependent on the displayed information
+            please use the linenumber panel as reference.
+            - If the size is not dependent on the displayed information please
+            use the debugger panel as reference.
+            - If your panel is in a floating position please use the
+            indentationguides as reference.
```

Comment:
If your panel is in a floating position, please use the IndentationGuide one as reference.